### PR TITLE
Reorganize closure tests to prepare for multiple typescript versions

### PR DESCRIPTION
**Pull request by julienp**
# Description

In preparation of https://github.com/pulumi/pulumi/issues/15735 we make the closure tests proper integration tests so that we can run them with different typescript versions. Move each test to its own folder instead of one large file.

This PR only changes tests, and does not touch any of the function serialisation code.

Some snapshots had to be updated for indentation changes.

Hidden after all the test cases is the test script [sdk/nodejs/tests/runtime/testdata/closure-tests/test.ts](https://github.com/pulumi/pulumi/blob/dfcc953c08051555ba1fa363b0694d195cfcc27a/sdk/nodejs/tests/runtime/testdata/closure-tests/test.ts)

Verified that tests run in CI https://github.com/pulumi/pulumi/actions/runs/8389170587/job/22975068167?pr=15753

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: sdk/nodejs/tsconfig.json [hunk 1]
```diff
@@ -99,10 +99,9 @@
         "tests/unwrap.spec.ts",
         "tests/util.ts",
         "tests/runtime/asyncIterableUtil.spec.ts",
-        "tests/runtime/closureLoader.spec.ts",
```
Comment (Frassle) [inline sdk/nodejs/tsconfig.json:102]: Can this file be deleted?

File: sdk/nodejs/tests/runtime/closure-integreation-tests.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        stdio: "inherit",
+    });
+
+    tmpDir.removeCallback();
+}
+
+async function main() {
+    for (const [ts, typesNode] of [
+        ["~3.8.3", "^17"],
+        // ["^4.9.5", "^20"],
```
Comment (Frassle) [inline sdk/nodejs/tests/runtime/closure-integreation-tests.ts:105]: In prep for supporting the other versions?

File: sdk/nodejs/tsconfig.json (hunk 1 above)
Comment (julienp) [reply sdk/nodejs/tsconfig.json:102]: Weird. It is deleted, but Git flags it as rename, guess there's just enough similarity to trip it up
```
renamed: sdk/nodejs/tests/runtime/closureLoader.spec.ts ⟶   sdk/nodejs/tests/runtime/testdata/closure-tests/cases/152-Capture-config-created-on-the-inside/deploymentOnlyModule/index.ts
```

File: sdk/nodejs/tests/runtime/closure-integreation-tests.ts (hunk 2 above)
Comment (julienp) [reply sdk/nodejs/tests/runtime/closure-integreation-tests.ts:105]: Yes, in the follow up PR I make use of this here https://github.com/pulumi/pulumi/pull/15761/files#diff-641029ea7bb84996df592ea47b810b71ab72bedf625235a56a24a438b5fa4b64

I can tidy up the PR here if you prefer.

**Review by julienp — COMMENTED** [review]
While working on the follow up I realised these tests don't run on windows/macOS. I added a new commit for this https://github.com/pulumi/pulumi/pull/15753/commits/435acf9b27b7c61fa04966a11e6a32f46ce2fb84

Wouldn't be surprised if there's going to be some line ending issues.

**Comment by julienp** [comment]
The tests now run on windows and macOS 

https://github.com/pulumi/pulumi/actions/runs/8420171003/job/23054623412?pr=15753#step:38:313

https://github.com/pulumi/pulumi/actions/runs/8420171003/job/23054621421?pr=15753#step:38:312

@Frassle I think this is good to go now

**Review by Frassle — APPROVED** [review]

**Pull request merged**